set.go

```diff
@@ -0,0 +1,68 @@
+package multihash
+
+// Set is a set of Multihashes, holding one copy per Multihash.
+type Set struct {
+	set map[string]struct{}
+}
+
+// NewSet creates a new set correctly initialized.
+func NewSet() *Set {
+	return &Set{
+		set: make(map[string]struct{}),
+	}
+}
+
+// Add adds a new multihash to the set.
+func (s *Set) Add(m Multihash) {
+	s.set[string(m)] = struct{}{}
+}
+
+// Len returns the number of elements in the set.
+func (s *Set) Len() int {
+	return len(s.set)
+}
+
+// Has returns true if the element is in the set.
+func (s *Set) Has(m Multihash) bool {
+	_, ok := s.set[string(m)]
+	return ok
+}
+
+// Visit adds a multihash only if it is not in the set already.  Returns true
+// if the multihash was added (was not in the set before).
+func (s *Set) Visit(m Multihash) bool {
+	// Do things directly here to avoid []byte -> string operations
+	str := string(m)
```

Comment:
I believe we need `s.set[string(m)]` to get this optimization: https://github.com/golang/go/wiki/CompilerOptimizations#map-lookup-by-byte.

Reply:
ah thanks!